src/utils/ex00_base.cairo

```diff
@@ -4,14 +4,32 @@
 // such as `validate_exercise`, `distribute_points`
 ////////////////////////////////
 
+use starknet::ContractAddress;
 
-#[contract]
+#[starknet::interface]
+trait Ex00BaseTrait<T> {
+    fn has_validated_exercise(ref self:T, account: ContractAddress) -> bool ;
+    fn ex_initializer(ref self:T);
+
+    fn tderc20_address(self: @T) -> ContractAddress ;
+    fn players_registry(self: @T) -> ContractAddress ;
+    fn workshop_id(self: @T) -> u128 ;
+    fn exercise_id(self: @T) -> u128 ;
+
+    // fn validate_exercise(ref self:T, account: ContractAddress) ;
+    // fn distribute_points(ref self: ContractState, to: ContractAddress, amount: u128) {
+    // fn update_class_hash_by_admin(ref self:T, class_hash_in_felt: felt252) ;
+
+
+}
+
+#[Starknet::contract]
 mod Ex00Base {
     // Core Library Imports
     use starknet::get_caller_address;
     use zeroable::Zeroable;
     use starknet::ContractAddress;
-    use starknet::ContractAddressZeroable;
+    // use starknet::ContractAddressZeroable;
```

Comment:
Consider using this trait (starknet::contract_address::ContractAddressZeroable) in place of commenting out thee line.

https://book.cairo-lang.org/appendix-05-most-common-types-and-traits.html#:~:text=starknet%3A%3Acontract_address%3A%3AContractAddressZeroable